### ISSUE TITLE
Unexpected/misleading IDE diagnostics for user defined assignment operators

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#78279]
```
class C1
{
    // CA1822 Member 'op_Increment' does not access instance data and can be marked as static
    public void operator ++() 
    {
    }

    // CA1822 Member 'op_SubtractionAssignment' does not access instance data and can be marked as static
    // IDE0060 Remove unused parameter 'i'
    public void operator -=(int i)
    {
    }
}
```
